### PR TITLE
Close #21379: Move map base z to constant

### LINKED ISSUE #21379: Remove or Expose gMapBaseZ
#21370 moved gMapBaseZ into the gamestate structure. But nothing ever sets this to a value other than 7 (except possibly from an s6 file). This issue is to discuss if we should either:

1. Change it to a constant `7` or
2. Expose it in the Ui so users can modify it.

## What does gMapBaseZ do:
It controls which land height counts as 0 metres/feet. This can be seen when viewing height labels and watching the altitude graph for a roller coaster. Purely cosmetic.

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Move the MapBaseZ gamestate variable to a constant as it's value never gets changed.
For https://github.com/OpenRCT2/OpenRCT2/issues/21379

### CODE REVIEW

File: src/openrct2/rct2/S6Importer.cpp [hunk 1]
```diff
@@ -410,7 +410,6 @@ namespace RCT2
             // rct1_water_colour
             // Pad01358842
             ImportResearchList(gameState);
-            gameState.MapBaseZ = _s6.MapBaseZ;
```
Comment (Harry-Hopkinson) [inline src/openrct2/rct2/S6Importer.cpp:413]: The MapBaseZ variable is always the same value so this is not needed.

**Comment by ZehMatt** [comment]
I don't think that we came to a conclusion on that discussion.

**Comment by Harry-Hopkinson** [comment]
Sorry, I  seemed to have jumped the gun. Should I close this PR or leave it open in case that they want to move the variable to a constant?

**Comment by Gymnasiast** [comment]
Let’s leave this open for a bit.

**Comment by Gymnasiast** [comment]
We left the discussion open for a bit longer, and since no objections came forward, I think we can proceed with this.

Could you rename the `MAPBASEZ` constant to `kMapBaseZ`?

**Comment by Harry-Hopkinson** [comment]
I have renamed it to kMapBaseZ.

File: src/openrct2/world/Map.h [hunk 2]
```diff
@@ -20,6 +20,7 @@ constexpr uint8_t MINIMUM_LAND_HEIGHT = 2;
 constexpr uint8_t MAXIMUM_LAND_HEIGHT = 254;
 constexpr uint8_t MINIMUM_WATER_HEIGHT = 2;
 constexpr uint8_t MAXIMUM_WATER_HEIGHT = 254;
+constexpr uint8_t kMapBaseZ = 7;
```
Comment (Gymnasiast) [inline src/openrct2/world/Map.h:23]:
```suggestion
/**
 * The land height that counts as 0 metres/feet for the land height labels and altitude graphs.
 */
constexpr uint8_t kMapBaseZ = 7;
```

**Review by Gymnasiast — APPROVED** [review]

File: src/openrct2/rct2/RCT2.h [hunk 3]
```diff
@@ -972,7 +972,7 @@ namespace RCT2
         uint8_t RCT1WaterColour;
         uint8_t Pad01358842[2];
         RCT12ResearchItem ResearchItems[Limits::MaxResearchItems];
-        uint16_t MapBaseZ;
+        uint16_t kMapBaseZ;
```
Comment (tupaschoal) [inline src/openrct2/rct2/RCT2.h:975]: Shouldn't this have been removed instead?

Comment (AaronVanGeffen) [reply src/openrct2/rct2/RCT2.h:975]: As understand it, it's part of the original save format, right?

Comment (tupaschoal) [reply src/openrct2/rct2/RCT2.h:975]: You mean it needs to be kept so it reads things in order? If that is the case, then we should rename or comment that this was made unused

Comment (Gymnasiast) [reply src/openrct2/rct2/RCT2.h:975]: We didn’t do that for all the other cases, though. The variable should be reverted, but a comment is not necessary.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Overlooked the accidental change in RCT2.h, please fix that.

**Comment by Harry-Hopkinson** [comment]
Fixed now, not sure how I missed that...

**Review by tupaschoal — APPROVED** [review]

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**